internal/loadgen/eventhandler.go

```diff
@@ -38,7 +42,34 @@ type EventHandlerParams struct {
 	RewriteTransactionNames   bool
 	RewriteTransactionTypes   bool
 	RewriteTimestamps         bool
-	Headers                   map[string]string
+	// Headers contains HTTP headers shipped with all requests.
+	// NOTE: these headers are not sanitized in logs.
+	Headers map[string]string
+}
+
+func (e EventHandlerParams) MarshalLogObject(enc zapcore.ObjectEncoder) error {
+	// NOTE: Logger is ignored.
+	enc.AddString("path", e.Path)
+	enc.AddString("url", e.URL)
+	enc.AddString("token", "REDACTED")
+	enc.AddString("api_key", "REDACTED")
+	// FIXME: add Limiter.
+	// FIXME: add Rand.
+	enc.AddBool("ignore_errors", e.IgnoreErrors)
+	enc.AddBool("rewrite_ids", e.RewriteIDs)
+	enc.AddBool("rewrite_service_names", e.RewriteServiceNames)
+	enc.AddBool("rewrite_ids", e.RewriteServiceNodeNames)
+	enc.AddBool("rewrite_ids", e.RewriteServiceTargetNames)
+	enc.AddBool("rewrite_ids", e.RewriteSpanNames)
+	enc.AddBool("rewrite_ids", e.RewriteTransactionNames)
+	enc.AddBool("rewrite_ids", e.RewriteTransactionTypes)
+	enc.AddBool("rewrite_ids", e.RewriteTimestamps)
```

Comment:
Did you mean to rename the keys here?
AFAICT we're adding multiple bool entries on the same "rewrite_ids" field?
I am not very familiar with the library so if it's on purpose, feel free to discard the comment

Reply:
No, this copy paste failure is on me.

Reply:
Fixed in [`29d86c6` (#47)](https://github.com/elastic/apm-perf/pull/47/commits/29d86c694da798d5f5b6e19bb86f44de9857e027)